src/serlio/prtMaterial/ArnoldMaterialNode.cpp

```diff
@@ -0,0 +1,575 @@
+
+	sb.declString(L"$shadingGroup");
+	sb.declString(L"$shaderNode");
+	sb.declString(L"$mapFile");
+	sb.declString(L"$mapNode");
+	sb.declString(L"$displacementNode");
+	sb.declString(L"$colorMapBlendNode");
+	sb.declString(L"$dirtMapBlendNode");
+	sb.declString(L"$opacityMapBlendNode");
+	sb.declString(L"$specularMapBlendNode");
+	sb.declString(L"$emissiveMapBlendNode");
+	sb.declString(L"$roughnessMapBlendNode");
+	sb.declString(L"$metallicMapBlendNode");
+	sb.declString(L"$uvTrafoNode");
+
+	for (adsk::Data::Handle& inMatStreamHandle : *inMatStream) {
+		if (!inMatStreamHandle.hasData())
+			continue;
+
+		if (!inMatStreamHandle.usesStructure(*fStructure))
+			continue;
+
+		MaterialInfo matInfo(inMatStreamHandle);
+
+		//material with same metadata already exists?
+		auto matchingMatIt = std::find_if(existingMaterialInfos.begin(), existingMaterialInfos.end(), [&matInfo](auto& p) {
+			return matInfo.equals(p.second);
+		});
+
+		MObject matchingMaterial = MObject::kNullObj;
+		if (matchingMatIt != existingMaterialInfos.end()) {
+			matchingMaterial = matchingMatIt->first;
+		}
+
+		if (!inMatStreamHandle.setPositionByMemberName(gPRTMatMemberFaceStart.c_str())) {
+			continue;
+		}
+		const int faceStart = *inMatStreamHandle.asInt32();
+
+		if (!inMatStreamHandle.setPositionByMemberName(gPRTMatMemberFaceEnd.c_str())) {
+			continue;
+		}
+		const int faceEnd = *inMatStreamHandle.asInt32();
+
+		if (matchingMaterial != MObject::kNullObj) {
+			MFnDependencyNode matchingMaterialDependencyNode(matchingMaterial, &status);
+			MCHECK(status);
+			MString matchingMaterialNodeName = matchingMaterialDependencyNode.name(&status);
+			MCHECK(status);
+			std::wstring matchingMaterialName(matchingMaterialNodeName.asWChar());
+			// TODO: try to find a more reliable way than comparing node names
+			const size_t idx = matchingMaterialName.find_last_of(L"Sh");
+			if (idx != std::wstring::npos) {
+				matchingMaterialName[idx] = L'g';
+			}
+			sb.setsAddFaceRange(matchingMaterialName, meshName.asWChar(), faceStart, faceEnd);
+			continue;
+		}
+
+		//get unique name
+		std::wstring shaderName(L"serlioGeneratedArnoldMaterialSh");
+		std::wstring shadingGroupName(L"serlioGeneratedArnoldMaterialSg");
+
+		int shIdx = 0;
+		while (shaderNames.find(shaderName + std::to_wstring(shIdx)) != shaderNames.end()) {
+			shIdx++;
+		}
+		shaderName += std::to_wstring(shIdx);
+		shadingGroupName += std::to_wstring(shIdx);
+		shaderNames.insert(shaderName);
+
+		buildMaterialShaderScript(sb, matInfo, shaderName, shadingGroupName, meshName.asWChar(), faceStart, faceEnd);
+	}
+
+	sb.execute();
+
+	return MStatus::kSuccess;
+}
+
+void ArnoldMaterialNode::buildMaterialShaderScript(MELScriptBuilder& sb,
+												   const MaterialInfo& matInfo,
+												   const std::wstring& shaderName,
+												   const std::wstring& shadingGroupName,
+												   const std::wstring& meshName,
+												   const int faceStart,
+												   const int faceEnd) {

```

Comment:
funny indentation

Reply:
The indentation is correct for a tab width of 4. Github uses a tab width of 8.

Since this project is still quite small, I'd propose to define a formatting style using `clang-format` and even strictly enforce this style in CI. Concerning indentation, I would be strongly in favor of always using spaces to avoid exactly this tab width issue.

Reply:
* clang-format fine for me
* but tab for indent and spaces for alignment would be my preference

Reply:
i've opened #37 for clang-format